enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,407 @@
+        mode: 0644
+        overwrite: true
+        path: /etc/nmstate/openshift/worker-1.yaml
+```
+
+The filenames must correspond to the hostname of each node as that will be
+used to map the configuration to the node. A separate manifest will need
+to be provided for masters and workers, which is typical when working with
+machine-configs.
+
+When the new service runs, it will copy the appropriate YAML file (based on
+the hostname of the node) to /etc/nmstate so it will be applied by the normal
+system NMState service. If there is already a file in /etc/nmstate with the
+same name, the service will fail in order to avoid overwriting important
+system configuration.
+
+Alternatively, there will be a mechanism to deploy a single cluster-wide
+configuration. That might look like:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 10-br-ex-master
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,aW50ZXJmYWNlczoKLSBuYW1lOiBlbnAyczAKICB0eXBlOiBldGhlcm5ldAogIHN0YXRlOiB1cAogIGlwdjQ6CiAgICBlbmFibGVkOiBmYWxzZQogIGlwdjY6CiAgICBlbmFibGVkOiBmYWxzZQotIG5hbWU6IGJyLWV4CiAgdHlwZTogb3ZzLWJyaWRnZQogIHN0YXRlOiB1cAogIGNvcHktbWFjLWZyb206IGVucDJzMAogIGlwdjQ6CiAgICBlbmFibGVkOiBmYWxzZQogICAgZGhjcDogZmFsc2UKICBpcHY2OgogICAgZW5hYmxlZDogZmFsc2UKICAgIGRoY3A6IGZhbHNlCiAgYnJpZGdlOgogICAgcG9ydDoKICAgIC0gbmFtZTogZW5wMnMwCiAgICAtIG5hbWU6IGJyLWV4Ci0gbmFtZTogYnItZXgKICB0eXBlOiBvdnMtaW50ZXJmYWNlCiAgc3RhdGU6IHVwCiAgaXB2NDoKICAgIGVuYWJsZWQ6IHRydWUKICAgIGRoY3A6IHRydWUKICBpcHY2OgogICAgZW5hYmxlZDogZmFsc2UKICAgIGRoY3A6IGZhbHNl
+        mode: 0644
+        overwrite: true
+        path: /etc/nmstate/openshift/cluster.yaml
+```
+
+The only difference is there will be a single file applied to each node of the
+specified role. The name of the file must be `cluster.yaml`.
+If there are both node-specific and cluster-wide configuration
+files present, the node-specific one will take precedence. Again, separate
+manifests will be needed for each role because of how MCO works.
+
+When this mechanism is in use, a flag will be set to indicate to configure-ovs
+that it should not attempt to configure br-ex.
+
+Day 2 changes will be handled by Kubernetes-NMState. For the initial
+implementation, a limited set of changes will be supported. This is because
+some operations will be disruptive and we don't currently have any way to
+orchestrate such a change. To begin, we intend to support the following day 2
+changes:
+
+- DNS updates
+- MTU updates
```

Comment:
There is a very little nit handling on configure-ovs tied to the MTU migration procedure:

https://github.com/openshift/machine-config-operator/blob/204767253e30608b5b7fd70ad1ace02ba1d64b46/templates/common/_base/files/configure-ovs-network.yaml#L880-L884

Reply:
Thanks for the heads up. I will make sure this is handled in the implementation patch.